xclim/core/calendar.py

```diff
@@ -337,7 +340,7 @@ def _convert_datetime(
     datetime: Union[pydt.datetime, cftime.datetime],
     new_doy: Optional[Union[float, int]] = None,
     calendar: str = "default",
-):
+) -> Union[cftime.datetime, pd.DatetimeIndex, np.int]:
```

Comment:
```suggestion
) -> Union[cftime.datetime, pydt.datetime, np.int]:
```
Would be more exact.